skcosmo/pcovr/pcovr.py

```diff
@@ -0,0 +1,313 @@
+            \\mathbf{X}\\right)^{-\\frac{1}{2}} \\mathbf{X}^T
+            \\mathbf{\\hat{Y}}\\mathbf{\\hat{Y}}^T \\mathbf{X} \\left(\\mathbf{X}^T
+            \\mathbf{X}\\right)^{-\\frac{1}{2}}\\right)
+
+        where
+
+        .. math::
+
+            \\mathbf{P}_{XT} = (\\mathbf{X}^T \\mathbf{X})^{-\\frac{1}{2}}
+                                \\mathbf{U}_\\mathbf{\\tilde{C}}^T
+                                \\mathbf{\\Lambda}_\\mathbf{\\tilde{C}}^{\\frac{1}{2}}
+
+        .. math::
+
+            \\mathbf{P}_{TX} = \\mathbf{\\Lambda}_\\mathbf{\\tilde{C}}^{-\\frac{1}{2}}
+                                \\mathbf{U}_\\mathbf{\\tilde{C}}^T
+                                (\\mathbf{X}^T \\mathbf{X})^{\\frac{1}{2}}
+
+        .. math::
+
+            \\mathbf{P}_{TY} = \\mathbf{\\Lambda}_\\mathbf{\\tilde{C}}^{-\\frac{1}{2}}
+                               \\mathbf{U}_\\mathbf{\\tilde{C}}^T (\\mathbf{X}^T
+                               \\mathbf{X})^{-\\frac{1}{2}} \\mathbf{X}^T
+                               \\mathbf{Y}
+
+        """
+
+        Ct, iCsqrt = pcovr_covariance(
+            mixing=self.mixing,
+            X_proxy=X,
+            Y_proxy=Yhat,
+            rcond=self.tol,
+            return_isqrt=True,
+        )
+        Csqrt = np.linalg.inv(iCsqrt)
+
+        v, U = self._eig_solver(Ct)
+        S = v ** 0.5
+        S_inv = np.linalg.pinv(np.diagflat(S))
+
+        self.singular_values_ = S.copy()
+        self.explained_variance_ = (S ** 2) / (X.shape[0] - 1)
+        self.explained_variance_ratio_ = (
+            self.explained_variance_ / self.explained_variance_.sum()
+        )
+
+        self.pxt_ = np.linalg.multi_dot([iCsqrt, U, np.diagflat(S)])
+        self.ptx_ = np.linalg.multi_dot([S_inv, U.T, Csqrt])
+        self.pty_ = np.linalg.multi_dot([S_inv, U.T, iCsqrt, X.T, Yhat])
+
+    def _fit_structure_space(self, X, Yhat, W):
+        """
+        In sample-space PCovR, the projectors are determined by:
+
+        .. math::
+
+            \\mathbf{\\tilde{K}} = \\alpha \\mathbf{X} \\mathbf{X}^T +
+            (1 - \\alpha) \\mathbf{\\hat{Y}}\\mathbf{\\hat{Y}}^T
+
+        where
+
+        .. math::
+
+            \\mathbf{P}_{XT} = \\left(\\alpha \\mathbf{X}^T + (1 - \\alpha)
+                               \\mathbf{W} \\mathbf{\\hat{Y}}^T\\right)
+                               \\mathbf{U}_\\mathbf{\\tilde{K}}
+                               \\mathbf{\\Lambda}_\\mathbf{\\tilde{K}}^{-\\frac{1}{2}}
+
+        .. math::
+
+            \\mathbf{P}_{TX} = \\mathbf{\\Lambda}_\\mathbf{\\tilde{K}}^{-\\frac{1}{2}}
+                                \\mathbf{U}_\\mathbf{\\tilde{K}}^T \\mathbf{X}
+
+        .. math::
+
+            \\mathbf{P}_{TY} = \\mathbf{\\Lambda}_\\mathbf{\\tilde{K}}^{-\\frac{1}{2}}
+                               \\mathbf{U}_\\mathbf{\\tilde{K}}^T \\mathbf{Y}
+
+        """
+
+        Kt = pcovr_kernel_distance(mixing=self.mixing, X_proxy=X, Y_proxy=Yhat)
```

Comment:
Can we maybe call this `pcovr_kernel` instead? To me, a "kernel distance" is the actual distance, i.e., d(a, b) = K(a, a) + K(b, b) - 2*K(a, b)

Reply:
I'm generally in favor of that as well. The name was suggested by @Luthaf in the selection PR